core/blockchain/blockchain_gateway.py

```diff
@@ -1,164 +1,155 @@
-
-    def listen_terminate(self) -> None:
+        logging.info("Starting cron...")
+        self._listen_as_event(period_in_mins, 
+                self._handle_new_session_creation,
+                self._filter_new_session)
+   
+    def stop_cron(self) -> None:
+        """
+        Stop the CRON method.
         """
-        Polls blockchain to see whether to terminate
+        self.event.set()
+        logging.info("Cron stopped!")
+  
+    def reset(self) -> None:
         """
-        def filter(tx):
-            return tx.get('content') is None
-        self.filter_set(filter, self.handle_terminate)
-    
\ No newline at end of file
+        Reset the gateway
+        """
+        self.event = Event()
+        self.state = []
+        logging.info("Gateway reset!")
+    
+    # Private methods to manage listening
+
+    def _update_local_state(self, global_state_wrapper: dict) -> None:
+        """
+        Helper function to update the local state with freshly downloaded global state.
+        """
+        self.state = global_state_wrapper.get(TxEnum.MESSAGES.name, {})
+    
+    def _listen(self, callback: Callable, 
+                event_filter: Callable) -> Tuple[Callable, Callable]:
+        """
+        Fetches the global state.
+        Passes the global state to a filter to see all relevant transactions.
+        Updates local state.
+        If any relevant transactions found, returns the callback result.
+        Else, returns the arguments it was passed.
+        """
+        global_state_wrapper = get_global_state(self.host, self.port, self.timeout)
+        filtered_diffs = filter_diffs(global_state_wrapper, self.state, event_filter)
+        self._update_local_state(global_state_wrapper)
+        if filtered_diffs:
+            return callback(filtered_diffs)
+        else:
+            return callback, event_filter
+   
+    def _listen_as_event(self, 
+                        period_in_mins: float, 
+                        callback: Callable, 
+                        event_filter: Callable) -> None:
+        """
+        Trigger above method every period.
+        """
+        new_callback, event_filter = self._listen(callback, event_filter)
+        if not self.event.is_set():
+            Timer(
+                period_in_mins * 60,
+                self._listen_as_event,
+                [period_in_mins, new_callback, event_filter]
+            ).start()
+
+    def _handle_new_session_creation(self, txs: list) -> Tuple[Callable, Callable]:
+        """
+        Maps the handler onto all relevant transactions.
+        Then returns the next handler and filter.
+        """
+        def handler(tx):
+            assert TxEnum.KEY.name in tx
+            key = tx.get(TxEnum.KEY.name)
+            value = tx.get(TxEnum.CONTENT.name)
+            self.communication_manager.inform(RawEventTypes.NEW_SESSION.name, ipfs_to_content(self.client, value))
+        list(map(handler, txs))
```

Comment:
This isn't assigned to anything, was this intentional?

Reply:
Yes